src/PJ_rpoly.c

```diff
@@ -34,6 +38,35 @@ static XY s_forward (LP lp, PJ *P) {           /* Spheroidal, forward */
 }
 
 
+static LP s_inverse (XY xy, PJ *P) {           /* Spheroidal, inverse */
+    LP lp = {0.0,0.0};
+    struct pj_opaque *Q = P->opaque;
+    double B, dphi, tp;
+    int i;
+
+    if (fabs(xy.y = P->phi0 + xy.y) <= TOL) {
```

Comment:
That was going to be my next comment. I basically tried to match the style of PJ_poly.c (american polyconic) for the sake of consistency, so you can see exactly how the two implementations are different, but I agree this is old-fashioned C and difficult to read. If we were compiling this with gcc -Wall it certainly wouldn't fly.

Other than the assignments in the if statements, I have several other issues with the code:

- `Q->fxa` and Q->fxb` set in `PJ *PROJECTION()` and used in the forward calculations. It's confusing and makes it hard to see how the equations translate to code.
- I don't like the do while loop for the Newton-Raphson iteration.
- Should I replace the `I_ERROR` macro with `pj_ctx_set_errno`?

So if I make changes here for code readability, I feel like I should update pj_poly.c as well for consistency. Or do you prefer minimal changes to existing code? Question of philosophy...